### PR TITLE
[libunwind] new port

### PR DESCRIPTION
**Pull request by Melnytskyi**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by Melnytskyi** [comment]
@microsoft-github-policy-service agree

File: ports/libunwind/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+vcpkg_configure_make(
+    SOURCE_PATH ${SOURCE_PATH}
+    AUTOCONFIG
+)
+vcpkg_build_make(ENABLE_INSTALL)
+vcpkg_fixup_pkgconfig()
+
+
+file(INSTALL "${CMAKE_CURRENT_LIST_DIR}/usage" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")
+vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE")
```
Comment (jimwang118) [inline ports/libunwind/portfile.cmake:19]: A blank line needs to be added at the end of the file.

File: ports/libunwind/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,19 @@
+vcpkg_minimum_required(VERSION 2022-10-12)
+
+vcpkg_from_git(
+    OUT_SOURCE_PATH SOURCE_PATH
+    URL "https://github.com/libunwind/libunwind.git"
+    REF 24947191d61dda869e039e0414fe97e9f594acd5
+    HEAD_REF "v1.7-stable"
```
Comment (jimwang118) [inline ports/libunwind/portfile.cmake:7]:
```suggestion
    REPO libunwind/libunwind
    REF “v${VERSION}”
    HEAD_REF master
```

File: ports/libunwind/vcpkg.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "libunwind",
+  "version": "1.7",
```
Comment (jimwang118) [inline ports/libunwind/vcpkg.json:3]:
```suggestion
  "version": "1.7.2",
```

File: ports/libunwind/vcpkg.json [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  "supports": "linux",
+  "dependencies": [
+    {
+      "name": "vcpkg-cmake",
+      "host": true
+    },
+    {
+      "name": "vcpkg-cmake-config",
+      "host": true
+    }
```
Comment (jimwang118) [inline ports/libunwind/vcpkg.json:16]:
```suggestion
    }
```

**Review by jimwang118 — CHANGES_REQUESTED** [review]

**Comment by jimwang118** [comment]
Note: I will be converting your PR to draft status. The above suggested changes are only recommendations. If you are willing to adopt them, please click "ready for review" after making the modifications. If you do not wish to make any changes, please click "ready for review" directly. That way, I can be aware that you've responded since you can't modify the tags.

File: ports/libunwind/portfile.cmake (hunk 2 above)
Comment (Melnytskyi) [reply ports/libunwind/portfile.cmake:7]: `vcpkg_from_git` accepts full url to git
also, main branch has more commits, so i used stable branch(they keep versions in branches)

Comment (dg0yt) [reply ports/libunwind/portfile.cmake:7]: You should use `vcpkg_from_github` instead.

Comment (Melnytskyi) [reply ports/libunwind/portfile.cmake:7]: `vcpkg_from_github` download archives, libunwind repo not use archives feature instead they keep version in branch

Comment (dg0yt) [reply ports/libunwind/portfile.cmake:7]: You misunderstand what `vcpkg_from_github`, or GH, is doing. The archives are created on the fly. It doesn't matter if REF is a commit, tag or branch.

Comment (dg0yt) [reply ports/libunwind/portfile.cmake:7]: ... and they are using regular git tags for releases.

File: ports/libunwind/usage [hunk 5]
```diff
@@ -0,0 +1,6 @@
+The package libunwind can be imported via CMake FindPkgConfig module:
+
+    find_package(PkgConfig)
+    pkg_check_modules(libunwind REQUIRED IMPORTED_TARGET GLOBAL libunwind>=1.7.2)
```
Comment (dg0yt) [inline ports/libunwind/usage:4]:
```suggestion
    pkg_check_modules(libunwind REQUIRED IMPORTED_TARGET GLOBAL libunwind)
```

We don't want to update this file every time, and we don't know what the user actually needs.

File: ports/libunwind/portfile.cmake [hunk 6]
```diff
@@ -0,0 +1,19 @@
+vcpkg_minimum_required(VERSION 2022-10-12)
+
```
Comment (dg0yt) [inline ports/libunwind/portfile.cmake:2]:
```suggestion
```

File: ports/libunwind/portfile.cmake [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+    URL "https://github.com/libunwind/libunwind.git"
+    REF 24947191d61dda869e039e0414fe97e9f594acd5
+    HEAD_REF "v1.7-stable"
+)
+
+vcpkg_configure_make(
+    SOURCE_PATH ${SOURCE_PATH}
+    AUTOCONFIG
+)
+vcpkg_build_make(ENABLE_INSTALL)
```
Comment (dg0yt) [inline ports/libunwind/portfile.cmake:15]:
```suggestion
vcpkg_install_make()
```

**Review by dg0yt — CHANGES_REQUESTED** [review]

File: ports/libunwind/portfile.cmake [hunk 8]
```diff
@@ -0,0 +1,18 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO "libunwind/libunwind"
+    REF v1.7.2
```
Comment (dg0yt) [inline ports/libunwind/portfile.cmake:4]:
```suggestion
    REF "v${VERSION}"
```

File: ports/libunwind/portfile.cmake [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+    HEAD_REF "v1.7-stable"
+    SHA512 b560f45cfa8ca3a60b41779afbcd862860c0f7af4014298c0b22eaec8b39740349c2077a940fb0d235c8f7d6a567e00c5cb0a04c251053f90c51320616784fd2
+)
+
+vcpkg_configure_make(
+    SOURCE_PATH ${SOURCE_PATH}
```
Comment (dg0yt) [inline ports/libunwind/portfile.cmake:10]:
```suggestion
    SOURCE_PATH "${SOURCE_PATH}"
```

**Review by dg0yt — COMMENTED** [review]
The current state of the libunwind project is unclear.
Is the official main repo savannah or Github?
Is 1.7.2 an official release?

Someone else must approve

**Comment by Melnytskyi** [comment]
> The current state of the libunwind project is unclear. Is the official main repo savannah or Github? Is 1.7.2 an official release?
> 
> Someone else must approve

Official repo is Github, in savannah lastest update(1.5) refer to github

1.7.2 is in github releases

**Comment by dg0yt** [comment]
> Official repo is Github, in savannah lastest update(1.5) refer to github

Okay, so the move is part of the "latest news" from savannah. It is a pity that they weren't able to update all the outdated information on the webpage.

**Review by jimwang118 — APPROVED** [review]

**Comment by jimwang118** [comment]
Usage test pass with following triplets:

```
x64-linux
```

**Pull request merged**